optimade/models/structures.py

```diff
@@ -893,6 +896,32 @@ def check_anonymous_formula(cls, v):
 
         return v
 
+    @validator("chemical_formula_anonymous", "chemical_formula_reduced")
+    def check_reduced_formulae(cls, value, field):
+        if value is None:
+            return value
+
+        numbers = [n.strip() or 1 for n in re.split(r"[A-Z][a-z]*", value)]
+        # Need to remove leading 1 from split and convert to ints
+        numbers = [int(n) for n in numbers[1:]]
+
+        if 0 in numbers:
+            raise ValueError(
+                f"{field.name} {value!r} cannot contain chemical proportion of 0."
+            )
+
+        if sys.version_info[1] >= 9:
+            gcd = math.gcd(*numbers)
+        else:
+            gcd = reduce(math.gcd, numbers)
```

Comment:
Sure - here or when importing 👍

Reply:
I didn't really want to mess around with a `partial(reduce(gcd, ...))` etc. We'll be able to deprecate 3.8 in 3 years anyway ;)

We can do a profiling/optimisation pass at some point but I don't think we need to yet.